lib/ansible/module_utils/vmware.py

```diff
@@ -952,6 +952,11 @@ def get_vm(self):
             elif vms:
                 # Unique virtual machine found.
                 vm_obj = vms[0]
+        elif self.params['moid']:
+            objects = self.content.viewManager.CreateContainerView(self.content.rootFolder, [vim.VirtualMachine], True).view
```

Comment:
More efficient way is to make an object with the moId.  Try:
```
    vm_obj = VmomiSupport.templateOf('VirtualMachine')(self.params['moid'], si._stub)
```
Must faster and doesn't involve a search on the server.

Reply:
Nice one, I am not all familair with pyvmomi so massive thanks for the suggestion!